src/libraries/System.Runtime.InteropServices/tests/DllImportGenerator.UnitTests/Compiles.cs

```diff
@@ -310,27 +310,38 @@ public async Task ValidateSnippetsWithPreprocessorDefintions(string source, IEnu
             Assert.Empty(newCompDiags);
         }
 
-        public static IEnumerable<object[]> CodeSnippetsToCompileWithForwarder()
+        public static IEnumerable<object[]> CodeSnippetsToValidateFallbackForwarder()
         {
-            yield return new[] { CodeSnippets.UserDefinedEntryPoint };
-            yield return new[] { CodeSnippets.AllSupportedDllImportNamedArguments };
+            yield return new object[] { CodeSnippets.UserDefinedEntryPoint, TestTargetFramework.Net, true };
 
-            // Parameter / return types (supported in DllImportGenerator)
-            yield return new[] { CodeSnippets.BasicParametersAndModifiers<byte>() };
-            // Parameter / return types (not supported in DllImportGenerator)
-            yield return new[] { CodeSnippets.BasicParametersAndModifiers<string>() };
+            // Confirm that all unsupported target frameworks can be generated.
+            {
+                string code = CodeSnippets.BasicParametersAndModifiers<byte>(CodeSnippets.GeneratedDllImportAttributeDeclaration);
+                yield return new object[] { code, TestTargetFramework.Net5, false };
+                yield return new object[] { code, TestTargetFramework.Core, false };
+                yield return new object[] { code, TestTargetFramework.Standard, false };
+                yield return new object[] { code, TestTargetFramework.Framework, false };
+            }
+
+            // Confirm that all unsupported target frameworks fallback to a forwarder.
+            {
+                string code = CodeSnippets.BasicParametersAndModifiersWithCharSet<string>(CharSet.Unicode, CodeSnippets.GeneratedDllImportAttributeDeclaration);
+                yield return new object[] { code, TestTargetFramework.Net5, true };
+                yield return new object[] { code, TestTargetFramework.Core, true };
+                yield return new object[] { code, TestTargetFramework.Standard, true };
+                yield return new object[] { code, TestTargetFramework.Framework, true };
+            }
         }
 
         [ConditionalTheory]
-        [MemberData(nameof(CodeSnippetsToCompileWithForwarder))]
-        public async Task ValidateSnippetsWithForwarder(string source)
+        [MemberData(nameof(CodeSnippetsToValidateFallbackForwarder))]
+        public async Task ValidateSnippetsFallbackForwarder(string source, TestTargetFramework targetFramework, bool expectFallbackForwarder)
         {
-            Compilation comp = await TestUtils.CreateCompilation(source);
+            Compilation comp = await TestUtils.CreateCompilationWithReferenceAssemblies(source, targetFramework);
             TestUtils.AssertPreSourceGeneratorCompilation(comp);
 
             var newComp = TestUtils.RunGenerators(
                 comp,
-                new DllImportGeneratorOptionsProvider(useMarshalType: false, generateForwarders: true),
```

Comment:
You gon rid of the test for the `GenerateForwarders` option?

Reply:
Yeah. Is this something we want to validate?